src/Symfony/Component/HttpFoundation/BinaryFileResponse.php

```diff
@@ -164,6 +166,10 @@ public function setContentDisposition($disposition, $filename = '', $filenameFal
         if ('' === $filenameFallback && (!preg_match('/^[\x20-\x7e]*$/', $filename) || false !== strpos($filename, '%'))) {
             $encoding = mb_detect_encoding($filename, null, true);
 
+            if (false === $encoding) {
+                throw new \InvalidArgumentException('The filename encoding cannot be detected.');
```

Comment:
When `$filename` is not a "clean" utf-8 string, but have a broken encoding, this could break the output wherever the exception message will be used. Actually we had an exception from the mongodb related code not showing up in the first place, cause its message included the broken non utf-8 encoded filename which in turn caused an Exception in the ExceptionListener, while trying to generate a JSON response with an invalid utf-8 string. So I suggest not to include the filename here.